src/components/feature-elevation-profile/feature-elevation-profile.vue

```diff
@@ -0,0 +1,169 @@
+<script setup lang="ts">
+/**
+ * Display an elevation graph according to a given feature. If feature is undefined,
+ * a default image of a graph is displayed in place of the graph.
+ * This component may be used in Draw components, Measure Profile component, and Infos component.
+ */
+import {
+  computed,
+  getCurrentInstance,
+  onMounted,
+  ShallowRef,
+  shallowRef,
+  watch,
+  watchEffect,
+} from 'vue'
+import { useTranslation } from 'i18next-vue'
+import { Feature } from 'ol'
+import { Geometry } from 'ol/geom'
+
+import ElevationProfile from '@/components/common/graph/elevation-profile.vue'
+import { ProfileData } from '@/components/common/graph/elevation-profile'
+import useProfilePosition from '@/composables/map/profile-position.composable'
+import { useProfilePositionStore } from '@/stores/profile-position.store'
+import { DrawnFeature } from '@/services/draw/drawn-feature'
+import {
+  exportFeatureService,
+  TFeatExport,
+} from '@/services/export-feature/export-feature.service'
+import { transform } from 'ol/proj'
+import { PROJECTION_LUX } from '@/composables/map/map.composable'
+
+defineEmits<{
+  (e: 'close'): void
+}>()
+
+const props = defineProps<{
+  feature: DrawnFeature | undefined
+}>()
+
+const profilePosition = useProfilePosition(undefined)
+const profilePositionStore = useProfilePositionStore()
+const { t } = useTranslation()
+
+const hasCloseListener = computed(
+  () => getCurrentInstance()?.vnode?.props?.onClose
+) // NB. useAttrs() not working as close event is defined with "emit"
+const profileData: ShallowRef<ProfileData | undefined> = shallowRef(undefined)
+const profileDataLength = computed(() => profileData.value?.length || 0)
+const profileDataLastValue = computed(
+  () => profileData.value && profileData.value[profileDataLength.value - 1]
+)
+const elevationGain = computed(() => profileDataLastValue.value?.elevationGain)
+const elevationLoss = computed(() => profileDataLastValue.value?.elevationLoss)
+const cumulativeElevation = computed(
+  () => profileDataLastValue.value?.cumulativeElevation
+)
+const isLoading = computed(() => props.feature && !profileData.value)
+const isDrawing = computed(() => !props.feature)
+
+onMounted(() => {
+  props.feature &&
+    props.feature.getProfile().then(data => (profileData.value = data))
+})
+
+watch(
+  () => props.feature,
+  (feature, featureOld) => {
+    profileData.value = undefined // Force refresh the graph
+
+    if (feature && feature !== featureOld) {
+      feature.getProfile().then(data => (profileData.value = data))
+    }
+  }
+)
+
+watchEffect(() => {
+  if (props.feature && !props.feature.profileData) {
+    profileData.value = undefined // Force refresh the graph
+    props.feature?.getProfile().then(data => (profileData.value = data))
+  }
+})
```

Comment:
Are the three listeners necessary?

Reply:
No, you are right, i have removed the second one. First one is to init the call api, and the third one (now the second one) is to force refresh when line is modified.